src/DefaultCoreManager.cc

```diff
@@ -39,15 +40,14 @@ DefaultCoreManager::coreAvailable(int myCoreId) {
 }
 
 /**
- * Prevent further creations on and remove all threads from the most
- * recently acquired core, and schedule releaseCore() on it. Log an error
- * and abort the process if loss of a core results in no cores available
- * for general scheduling.
+ * Record the loss of a core and return its coreId.
  */
-void
+int
 DefaultCoreManager::coreUnavailable() {
     Lock guard(lock);
-    // TODO(hq6): Implement this function
+    int freeCoreId = sharedCores[sharedCores.size() - 1];
```

Comment:
May need to check sharedCores.size() > 0 before this line, in case that we mistakenly call this function. Return value should be valid coreId if success, and return -1 on error.

Reply:
Done.